getting-started/bookshelf/db_firestore.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"cloud.google.com/go/firestore"
+	"google.golang.org/api/iterator"
+)
+
+// firestoreDB persists books to Cloud Firestore.
+// See https://cloud.google.com/firestore/docs.
+type firestoreDB struct {
+	client *firestore.Client
+}
+
+// Ensure firestoreDB conforms to the BookDatabase interface.
+var _ BookDatabase = &firestoreDB{}
```

Comment:
This is completely okay as is, but consider moving it to a _test.go file since it's not a piece of code that is needed for functionality of the app.

Doing so causes a failure for `*firestoreDB` to implement the `BookDatabase` interface to be reported when you do `go test` rather than `go build`, which feels more fitting IMO.

Reply:
I think I like it in here - when `firestoreDB` stops implementing `BookDatabase`, you know right away in your editor, unlike when it's in the test file (for VS Code, at least).

Reply:
Won't it fail for `go test`, too? Package must compile for the tests to run.